server/components/searchQuery.js

```diff
@@ -1,14 +1,24 @@
 exports.emitSearchQuery = async (socket, inputQuery, allSymbols) => {
   try {
     const symbols = await allSymbols;
-    const filteredData = symbols.filter(
+
+    const filterData = symbols.filter(
       search =>
         search.symbol.toLowerCase().indexOf(inputQuery.toLowerCase()) !== -1 ||
         search.name.toLowerCase().indexOf(inputQuery.toLowerCase()) !== -1
     );
-    const topTen = filteredData.slice(0, 10);
 
-    socket.emit("companySymbols", topTen);
+    const dataSort = filterData.sort((a, b) => {
+      const aStart =
+        a.symbol.match(new RegExp("^" + inputQuery.toUpperCase())) || [];
```

Comment:
Perf: I'd take the `new RegExp("^" + inputQuery.toUpperCase())` out of this sort and re-use it, so it doesn't have to create a new RegExp object twice on every function invocation.